packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -62,26 +65,39 @@ export class HostedPluginSupport {
     @inject(PreferenceProviderProvider)
     protected readonly preferenceProviderProvider: PreferenceProviderProvider;
 
+    @inject(PreferenceServiceImpl)
+    private readonly preferenceServiceImpl: PreferenceServiceImpl;
+
+    @inject(PluginPathsService)
+    private readonly pluginPathsService: PluginPathsService;
+
+    @inject(StoragePathService)
+    private readonly storagePathService: StoragePathService;
+
+    @inject(WorkspaceService)
+    protected readonly workspaceService: WorkspaceService;
+
+    @inject(MonacoTextmateService)
+    protected readonly monacoTextmateService: MonacoTextmateService;
+
     private theiaReadyPromise: Promise<any>;
-    private frontendExtManagerProxy: PluginManagerExt;
-    private backendExtManagerProxy: PluginManagerExt;
+
+    protected readonly managers: PluginManagerExt[] = [];
```

Comment:
@evidolob @benoitf I've refactored this code to support multiple managers properly. Before one backend manager will receive events from the frontend. Are you fine with such breaking changes? I will update the changelog.

Reply:
I think refactoring and new feature should be separate commits or separate PR
moving from constructor injection to property injection is not directly linked to that PR so it makes a lot of extra changes for nothing.
I think this is kind of PR that I dislike the more when there is one change that is important but it's flooded by tons of  cleanup/style/user preferences that should be just for that.
It's like something can't be added without changing all the code which usually is not so true. 

BTW 'properly' is not meaningful. You should describe why it was limited and how you've removed that limitation so that people using that class are able to see what was the goal. This is more meaningful for me that a checklist of 100 items which are not describing the why, how and what.

besides that it's a great feature and I'm ok with the refactoring. But breaking change doesn't mean to mix enhancement/bugfixes with code cleanup/rewrite for its own style preference, etc the class in one step. It's so inconvenient to review.

Reply:
> I think refactoring and new feature should be separate commits or separate PR
moving from constructor injection to property injection is not directly linked to that PR so it makes a lot of extra changes for nothing.
I think this is kind of PR that I dislike the more when there is one change that is important but it's flooded by tons of cleanup/style/user preferences that should be just for that.
It's like something can't be added without changing all the code which usually is not so true.

Generally agree that PRs should be to the point. Not sure about refactoring PRs itself. It is fine if there is good set of regression tests, but if there are not it is an extra effort to test again. I would try to make additional explicit commits for refactoring next time or better to look in the test infrastructure next weeks.

> BTW 'properly' is not meaningful. You should describe why it was limited and how you've removed that limitation so that people using that class are able to see what was the goal. This is more meaningful for me that a checklist of 100 items which are not describing the why, how and what.

To be fair I've tried to describe it with: `Before one backend manager will receive events from the frontend.`. Maybe it was not clear: now all remote plugin host processes get notified about updated storage path and activation events. Agree that i had to put it in the description.